### PR TITLE
Introduce python serialization for ContainerBlueprint

### PR DESCRIPTION
**Pull request by jleibs**
### What
Cherry-picked over the Name / Visibillity changes from `andreas/serializable-spaceviewblueprint`, so we'll need to navigate merge conflicts there.

Standard implementation of serializers and a unit-test for them:

After talking to @Wumpf I moved some of the list-types up a level to simplify the serialization code, but this raised an issue with mismatched dimensions. Need to discuss if we want to remove this check (in light of the instance-key simplification), or push things back down to mono-components with internal lists.

### TODO

- [x] Dragging onto a Grid container breaks due to the mismatched dimensions between the contents and the row-shares.
  - Fixed by: https://github.com/rerun-io/rerun/pull/5399
```
[2024-03-04T23:10:53Z WARN  re_log::result_extensions] crates/re_viewport/src/container.rs:240 Failed to create Container blueprint.: Each cell must contain either 0, 1 or `num_instances` instances, but cell 'rerun.blueprint.components.RowShare' in '/container/439ad7ad-829c-4085-833b-eb30efae0810' holds 2 instances (expected 3)
```

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using newly built examples: [app.rerun.io](https://app.rerun.io/pr/5390/index.html)
  * Using examples from latest `main` build: [app.rerun.io](https://app.rerun.io/pr/5390/index.html?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [app.rerun.io](https://app.rerun.io/pr/5390/index.html?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/5390)
- [Docs preview](https://rerun.io/preview/6964d9643faba86a734746885f4a8792da3685f1/docs) 
- [Examples preview](https://rerun.io/preview/6964d9643faba86a734746885f4a8792da3685f1/examples) 
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

### CODE REVIEW

File: crates/re_types/definitions/rerun/blueprint/components/entities_determined_by_user.fbs [hunk 1]
```diff
@@ -11,9 +11,11 @@ namespace rerun.blueprint.components;
 
 /// Whether the space view entities were manually edited.
 table EntitiesDeterminedByUser (
+    "attr.arrow.transparent",
     "attr.rerun.scope": "blueprint",
     "attr.rust.derive": "Default, PartialEq, Eq, PartialOrd, Ord",
-    "attr.rust.repr": "transparent"
+    "attr.rust.repr": "transparent",
+    "attr.rust.tuple_struct"
```
Comment (Wumpf) [inline crates/re_types/definitions/rerun/blueprint/components/entities_determined_by_user.fbs:18]: I'm surprised we don't have a python alias defined here, I got mypy errors when I didn't put `bool` in there explicitely

File: crates/re_types/definitions/rerun/blueprint/components/included_content.fbs [hunk 2]
```diff
@@ -11,7 +11,7 @@ namespace rerun.blueprint.components;
 
 
 /// All the contents in the container.
-table IncludedContents (
+table IncludedContent (
```
Comment (Wumpf) [inline crates/re_types/definitions/rerun/blueprint/components/included_content.fbs:14]: Surprised this isn't arrow transparent

File: crates/re_types/definitions/rerun/blueprint/components/row_share.fbs [hunk 3]
```diff
@@ -9,12 +9,13 @@ namespace rerun.blueprint.components;
 
 // ---
 
-
-/// The layout shares of each column in the container.
-table ColumnShares (
+/// The layout share of a row in the container.
+table RowShare (
+    "attr.python.aliases": "float",
+    "attr.python.array_aliases": "npt.NDArray[np.float64]",
```
Comment (Wumpf) [inline crates/re_types/definitions/rerun/blueprint/components/row_share.fbs:15]: as above with ColumnShares

File: crates/re_types_builder/src/codegen/python/mod.rs [hunk 4]
```diff
@@ -200,8 +200,14 @@ impl ExtensionClass {
...
 
+            let scope = if let Some(scope) = obj.scope() {
+                format!("{scope}.")
+            } else {
+                String::new()
+            };
+
             let mandatory_docstring = format!(
-                r#""""Extension for [{name}][rerun.{kind}.{name}].""""#,
+                r#""""Extension for [{name}][rerun.{scope}{kind}.{name}].""""#,
```
Comment (Wumpf) [inline crates/re_types_builder/src/codegen/python/mod.rs:210]: ah good fix, I just yielded to the broken version so far

File: rerun_py/rerun_sdk/rerun/blueprint/components/column_share_ext.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+if TYPE_CHECKING:
+    from . import ColumnShareArrayLike
+
+
+class ColumnShareExt:
+    """Extension for [ColumnShare][rerun.blueprint.components.ColumnShare]."""
+
+    @staticmethod
+    def native_to_pa_array_override(data: ColumnShareArrayLike, data_type: pa.DataType) -> pa.Array:
+        array = np.asarray(data, dtype=np.float64).flatten()
```
Comment (Wumpf) [inline rerun_py/rerun_sdk/rerun/blueprint/components/column_share_ext.py:17]: why double here? fbs says 'float'

File: rerun_py/rerun_sdk/rerun/blueprint/components/row_share_ext.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+if TYPE_CHECKING:
+    from . import RowShareArrayLike
+
+
+class RowShareExt:
+    """Extension for [RowShare][rerun.blueprint.components.RowShare]."""
+
+    @staticmethod
+    def native_to_pa_array_override(data: RowShareArrayLike, data_type: pa.DataType) -> pa.Array:
+        array = np.asarray(data, dtype=np.float64).flatten()
```
Comment (Wumpf) [inline rerun_py/rerun_sdk/rerun/blueprint/components/row_share_ext.py:17]: as above with column

**Review by Wumpf — CHANGES_REQUESTED** [review]
some confusion around aliases and float/double. Otherwise looks good! Your test makes a little bit more sense than the ones I have - I should also feed in more arrays with the component type already provided.
We'll have to fix the grid stuff first ofc as well as you pointed out

File: crates/re_types/definitions/rerun/blueprint/components/column_share.fbs [hunk 7]
```diff
@@ -9,11 +9,14 @@ namespace rerun.blueprint.components;
 
 // ---
 
-/// The layout shares of each row in the container.
-table RowShares (
+
+/// The layout share of a column in the container.
+table ColumnShare (
+    "attr.python.aliases": "float",
+    "attr.python.array_aliases": "npt.NDArray[np.float64]",
```
[the remark this answers is not in the document]
Comment (jleibs) [reply crates/re_types/definitions/rerun/blueprint/components/column_share.fbs:16]: Bad copy-and-paste propagation

File: crates/re_types/definitions/rerun/blueprint/components/included_content.fbs (hunk 2 above)
Comment (jleibs) [reply crates/re_types/definitions/rerun/blueprint/components/included_content.fbs:14]: Components are always arrow transparent by default.

File: crates/re_types/definitions/rerun/blueprint/components/entities_determined_by_user.fbs (hunk 1 above)
Comment (jleibs) [reply crates/re_types/definitions/rerun/blueprint/components/entities_determined_by_user.fbs:18]: I think this is something that come over partially from the cherry-pick

**Pull request merged**